### PR TITLE
Add basic success/error metrics for components.

### PR DESCRIPTION
**Pull request by michelle192837**
Manually tested that these will log when the components are run.

### CODE REVIEW

File: pkg/updater/updater.go [hunk 1]
```diff
@@ -132,36 +186,7 @@ func Update(parent context.Context, client gcs.ConditionalClient, configPath gcs
...
-					}
-					log.Debug("Acquired update lock")
-				}
-				if err := updateGroup(ctx, log, client, &tg, *tgp); err != nil {
-					log.WithError(err).Error("Error updating group")
-				}
-				// run the garbage collector after each group to minimize
-				// extraneous memory usage.
-				runtime.GC()
+				update(ctx, client, mets, log, &tg, configPath, gridPrefix, updateGroup, write, generations)
```
Comment (fejta) [inline pkg/updater/updater.go:189]: This seems a lot cleaner than futzing around with `var errored bool`

```golang
if err := update(...); err != nil {
  mets.Errors.Add(1, "updater")
} else {
  mets.Successes.Add(1, "updater")
}
```

File: pkg/updater/updater.go [hunk 2]
```diff
@@ -109,8 +115,56 @@ func lockGroup(ctx context.Context, client gcs.ConditionalClient, path gcs.Path,
 	return gcs.Touch(ctx, client, path, generation, buf)
 }
 
+func update(ctx context.Context, client gcs.ConditionalClient, mets Metrics, log logrus.FieldLogger, tg *configpb.TestGroup, configPath gcs.Path, gridPrefix string, updateGroup GroupUpdater, write bool, generations map[string]int64) {
+	var errored bool
+	defer func() {
+		if errored {
+			if mets.Errors != nil {
+				mets.Errors.Add(1, "updater")
```
Comment (fejta) [inline pkg/updater/updater.go:123]: Seems kind of strange to make the caller remember to add an updater string. Seems like this should be embedded into either the mets or Errors structure.

File: pkg/updater/updater_test.go [hunk 3]
```diff
@@ -229,6 +229,7 @@ func TestUpdate(t *testing.T) {
 			err := Update(
 				ctx,
 				client,
+				Metrics{},
```
Comment (fejta) [inline pkg/updater/updater_test.go:232]: Can we unit test this new scenario? AKA successes should increment the success counter, errors should increment the error counter.

File: pkg/updater/updater.go [hunk 4]
```diff
@@ -109,8 +115,56 @@ func lockGroup(ctx context.Context, client gcs.ConditionalClient, path gcs.Path,
...
+			if mets.Successes != nil {
+				mets.Successes.Add(1, "updater")
+			}
+		}
+	}()
+	log = log.WithField("group", tg.Name)
+	log.Debug("Starting update")
+	tgp, err := testGroupPath(configPath, gridPrefix, tg.Name)
+	if err != nil {
+		log.WithError(err).Error("Bad path")
```
[the remark this answers is not in the document]
Comment (michelle192837) [reply pkg/updater/updater.go:135]: Fair enough, but it's in the original code so leaving it in for now.

File: pkg/updater/updater.go [hunk 5]
```diff
@@ -109,8 +115,56 @@ func lockGroup(ctx context.Context, client gcs.ConditionalClient, path gcs.Path,
...
+		return
+	}
+	if write && generations != nil {
+		if err := lockGroup(ctx, client, *tgp, generations[tg.Name]); err != nil {
+			var ok bool
+			switch ee := err.(type) {
+			case *googleapi.Error:
+				if ee.Code == http.StatusPreconditionFailed {
+					ok = true
+					log.Debug("Lost the lock race")
```
[the remark this answers is not in the document]
Comment (michelle192837) [reply pkg/updater/updater.go:145]: Good point, adding a TODO for now

**Review by fejta — APPROVED** [review]

**Pull request merged**